packages/fe/components/tooltip.vue

```diff
@@ -0,0 +1,89 @@
+<template>
+  <div :class="`tooltip-c align-${align}`">
+
+    <div class="tooltip-btn-wrapper">
+      <slot name="tooltip-btn" />
+    </div>
+
+    <div class="tooltip-box-wrapper">
+      <slot name="tooltip-box" />
+    </div>
+
+  </div>
+</template>
+
+<script>
+// ====================================================================== Export
+export default {
+  name: 'Tooltip',
+
+  props: {
+    align: {
+      type: String,
+      required: false,
+      default: 'center'
+    }
+  }
+}
+</script>
+
+<style lang="scss" scoped>
```

Comment:
Styling inside a renderless/abstract component should be suuuuper minimal. It should only include the absolute basic amount of styling to make the component functional. In most cases this is just some positioning, sizing and spacing. Like these guys:

```css
position: absolute;
top: 100%;
left: 50%;
transform: translateX(-50%);
```

All other styling should be applied outside the component. The reason for this is so that the abstracted component can be carried over to other projects without throwing errors (like undefined SCSS variables) and also so that we don't have to constantly remove previous-project styles and add current-project styles.

Take a look at the `select.vue` components for an example (although they're kinda complex). The select inside the forms module is the renderless version and the one in the top-level components directory is the custom-to-this-project version that contains most of the actual styling.

Can you create a new module called `zero` with a `components` directory inside it and move your new abstract Tooltip into it? And then create a new component in the project-level components directory also called `Tooltip` that uses the renderless tooltip internally?

Reply:
Oh 1 more thing → don't forget to add spaces between SCSS blocks 🙂